server/src/main/java/io/crate/types/GeoShapeType.java

```diff
@@ -65,6 +69,20 @@ public ValueIndexer<Map<String, Object>> valueIndexer(RelationName table,
                                                               Function<ColumnIdent, Reference> getRef) {
             return new GeoShapeIndexer(ref);
         }
+
+        @Override
+        public Map<String, Object> decode(ColumnIdent column, SourceParser sourceParser, byte[] bytes) {
+            try (StreamInput in = new ByteBufferStreamInput(ByteBuffer.wrap(bytes))) {
+                return GeoShapeType.INSTANCE.streamer().readValueFrom(in);
+            } catch (IOException ee) {
+                throw new UncheckedIOException(ee);
+            }
+        }
+
+        @Override
+        public boolean retrieveFromStoredFields() {
+            return true;
```

Comment:
maybe worth adding this comment as well?
```suggestion
            // there's a difference in precision between the stored value and the dv value
            // so when we're retrieving for display or exact comparisons, go to the stored
            // value
            return true;
```